backends/qualcomm/runtime/QnnExecuTorchBackend.cpp

```diff
@@ -163,15 +163,15 @@ Error QnnExecuTorchBackend::execute(
 
   for (int i = 0; i < input_tensors.size(); ++i) {
     input_tensors[i]->FillDataBuffer(
-        args[i]->toTensor().data_ptr(), true /* copy_data */);
+        args[i]->toTensor().const_data_ptr(), true /* copy_data */);
```

Comment:
Is this trying to write into `args[i]->toTensor()` or read from it? If writing into it then we need the mutable data pointer.

Reply:
For input tensors, we will only copy data and perform no further data processing. I think `const_data_ptr` might be fine.